cookiecutter/main.py

```diff
@@ -138,16 +138,12 @@ def cookiecutter(
     # except when 'no-input' flag is set
 
     with import_patch:
-        if context_for_prompting['cookiecutter']:
-            context['cookiecutter'].update(
-                prompt_for_config(context_for_prompting, no_input)
-            )
-        if "template" in context["cookiecutter"]:
-            nested_template = re.search(
-                r'\((.*?)\)', context["cookiecutter"]["template"]
-            ).group(1)
+        prompt_keys = list(context["cookiecutter"].keys())
+        has_nested_template = "template" in prompt_keys or "templates" in prompt_keys
+        if has_nested_template:
```

Comment:
This can be simplified using a set intersection:

```suggestion
        if {"template", "templates"} & context["cookiecutter"].keys():
```